File: eden/scm/edenscm/metrics.py
# Copyright (c) Meta Platforms, Inc. and affiliates.
#
# This software may be used and distributed according to the terms of the
# GNU General Public License version 2.

# metrics.py - generic metrics framework

from __future__ import absolute_import

from bindings import hgmetrics


class metrics(object):
    """Abstract base class for metrics"""

    def __init__(self, ui):
        self.ui = ui

    def gauge(self, key, value=1, entity=None):
        """If entity is None, log locally. Otherwise, send it to a global counter."""
        if entity is None:
            hgmetrics.incrementcounter(key, value)


def client(ui):
    """Returns the appropriate metrics module"""
    # @fb-only: from . import fb 

    # @fb-only: return fb.fbmetrics(ui) 
    return metrics(ui) # @oss-only

File: eden/scm/lib/commands/src/commands/debug/metrics.rs
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This software may be used and distributed according to the terms of the
 * GNU General Public License version 2.
 */

use clidispatch::ReqCtx;

use super::NoOpts;
use super::Repo;
use super::Result;

pub fn run(_ctx: ReqCtx<NoOpts>, _repo: &mut Repo) -> Result<u8> {
    hg_metrics::increment_counter("test_counter", 1);
    tracing::debug!(target: "test_trace", trace_key="trace-value");
    Ok(0)
}

pub fn aliases() -> &'static str {
    "debugmetrics"
}

pub fn doc() -> &'static str {
    "output test metrics from native command"
}

pub fn synopsis() -> Option<&'static str> {
    None
}
